parachains/runtimes/collectives/collectives-polkadot/src/ambassador/mod.rs

```diff
@@ -0,0 +1,248 @@
+	type RuntimeEvent = RuntimeEvent;
+	type Members = pallet_ranked_collective::Pallet<Runtime, AmbassadorCollectiveInstance>;
+	type Balance = Balance;
+	// Parameters are set by any of:
+	// - Root;
+	// - the FellowshipAdmin origin (i.e. token holder referendum);
+	// - a vote among all Head Ambassadors.
+	type ParamsOrigin = EitherOfDiverse<
+		EnsureXcm<IsVoiceOfBody<GovernanceLocation, FellowshipAdminBodyId>>,
+		EnsureHeadAmbassadorVoice,
+	>;
+	// Induction (creating a candidate) is by any of:
+	// - Root;
+	// - the FellowshipAdmin origin (i.e. token holder referendum);
+	// - a single Head Ambassador;
+	// - a vote among all senior members.
+	type InductOrigin = EitherOfDiverse<
+		EnsureXcm<IsVoiceOfBody<GovernanceLocation, FellowshipAdminBodyId>>,
+		EitherOfDiverse<
+			pallet_ranked_collective::EnsureMember<
+				Runtime,
+				AmbassadorCollectiveInstance,
+				{ ranks::HEAD_AMBASSADOR_TIER_5 },
+			>,
+			EnsureAmbassadorVoiceFrom<ConstU16<{ ranks::SENIOR_AMBASSADOR_TIER_3 }>>,
+		>,
+	>;
+	type ApproveOrigin = PromoteOrigin;
+	type PromoteOrigin = PromoteOrigin;
+	type EvidenceSize = ConstU32<65536>;
+}
+
+pub type AmbassadorSalaryInstance = pallet_salary::Instance2;
+
+parameter_types! {
+	// The interior location on AssetHub for the paying account. This is the Ambassador Salary
+	// pallet instance (which sits at index 64). This sovereign account will need funding.
+	pub Interior: InteriorMultiLocation = PalletInstance(74).into();
+}
+
+const USDT_UNITS: u128 = 1_000_000;
+
+/// [`PayOverXcm`] setup to pay the Ambassador salary on the AssetHub in USDT.
+pub type AmbassadorSalaryPaymaster = PayOverXcm<
+	Interior,
+	crate::xcm_config::XcmRouter,
+	crate::PolkadotXcm,
+	ConstU32<{ 6 * HOURS }>,
+	AccountId,
+	(),
+	ConvertToValue<UsdtAsset>,
+	AliasesIntoAccountId32<(), AccountId>,
+>;
+
+impl pallet_salary::Config<AmbassadorSalaryInstance> for Runtime {
+	type WeightInfo = weights::pallet_salary_ambassador_salary::WeightInfo<Runtime>;
+	type RuntimeEvent = RuntimeEvent;
+
+	#[cfg(not(feature = "runtime-benchmarks"))]
+	type Paymaster = AmbassadorSalaryPaymaster;
+	#[cfg(feature = "runtime-benchmarks")]
+	type Paymaster = crate::impls::benchmarks::PayWithEnsure<
+		AmbassadorSalaryPaymaster,
+		crate::impls::benchmarks::OpenHrmpChannel<ConstU32<1000>>,
+	>;
+	type Members = pallet_ranked_collective::Pallet<Runtime, AmbassadorCollectiveInstance>;
+
+	#[cfg(not(feature = "runtime-benchmarks"))]
+	type Salary = pallet_core_fellowship::Pallet<Runtime, AmbassadorCoreInstance>;
+	#[cfg(feature = "runtime-benchmarks")]
+	type Salary = frame_support::traits::tokens::ConvertRank<
+		crate::impls::benchmarks::RankToSalary<Balances>,
+	>;
+	// 15 days to register for a salary payment.
+	type RegistrationPeriod = ConstU32<{ 15 * DAYS }>;
+	// 15 days to claim the salary payment.
+	type PayoutPeriod = ConstU32<{ 15 * DAYS }>;
+	// Total monthly salary budget.
+	type Budget = ConstU128<{ 100_000 * USDT_UNITS }>;
```

Comment:
How is this distributed? Is it fully distributed amongst members (i.e. if there were only 1 member they would get 100%), or is a salary set per member, and it just cannot exceed this?

Reply:
this is the total `Budget` for salaries paid in one period.
every rank has a salary amount paid every period. it can be set via dispatchable call in fellowship core pallet.